src/fusiontensor/fusiontensoraxes.jl

```diff
@@ -0,0 +1,137 @@
+using BlockArrays: BlockArrays
+using GradedArrays:
+  GradedArrays,
+  AbstractGradedUnitRange,
+  AbstractSector,
+  TrivialSector,
+  dual,
+  sector_type,
+  trivial
+using TensorAlgebra: TensorAlgebra, AbstractBlockPermutation, BlockedTuple
+using TensorProducts: ⊗
+using TypeParameterAccessors: type_parameters
+
+# =======================================  Misc  ===========================================
+
+promote_sector_type(legs::Tuple) = promote_sector_type(legs...)
+
+function promote_sector_type(legs...)
+  # fuse trivial sectors to produce unified type
+  # avoid depending on GradedArrays internals
+  return sector_type(⊗(trivial.(legs)...))
+end
+
+promote_sectors(legs::NTuple{<:Any,<:AbstractGradedUnitRange}) = legs # nothing to do
+
+function promote_sectors(legs)
+  T = promote_sector_type(legs)
+  # fuse with trivial to insert all missing arguments inside each GradedAxis
+  # avoid depending on GradedArrays internals
+  s0 = trivial(T)
+  return map_sectors.(s -> only(sectors(to_gradedrange(tensor_product(s0, s)))), legs)
+end
+
+function promote_sectors(bt::BlockedTuple)
+  return BlockedTuple{blocklength(bt),blocklengths(bt)}(promote_sectors(Tuple(bt)))
+end
+
+# ====================================  Definitions  =======================================
+
+# TBD explicit axis type as type parameters?
+struct FusionTensorAxes{BT<:BlockedTuple{2}}
+  outer_axes::BT
+
+  function FusionTensorAxes{BT}(bt) where {BT}
+    @assert BT === typeof(promote_sectors(bt))
+    return new{BT}(bt)
+  end
+end
+
+# =====================================  Accessors  ========================================
+
+outer_axes(fta::FusionTensorAxes) = fta.outer_axes
```

Comment:
```suggestion
TensorAlgebra.BlockedTuple(fta::FusionTensorAxes) = fta.outer_axes
```
I think this interface is clearer since it is really converting the `FusionTensorAxes` to the equivalent `BlockedTuple`. The term `outer_axes` primarily makes sense for the FusionTensor itself but I think the interface of `FusionTensorAxes` should focus on treating it like a blocked tuple-like object.